redfish-core/lib/pcie_slots.hpp

```diff
@@ -29,6 +32,88 @@
 namespace redfish
 {
 
+inline void doLinkAssociatedDiskBackplaneToChassis(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& chassisId, const std::string& drivePath, size_t index,
+    const std::optional<std::string>& validChassisPath,
+    const std::vector<std::string>& assemblyList)
+{
+    if (!validChassisPath || assemblyList.empty())
+    {
+        BMCWEB_LOG_ERROR("Chassis not found");
+        messages::resourceNotFound(asyncResp->res, "Chassis", chassisId);
+        return;
+    }
+
+    auto it = std::find(assemblyList.begin(), assemblyList.end(), drivePath);
+    if (it == assemblyList.end())
+    {
+        BMCWEB_LOG_ERROR("Drive path {} not found in the assembly list",
+                         drivePath);
+        messages::internalError(asyncResp->res);
+    }
+
+    nlohmann::json::object_t item;
+    item["@odata.id"] = boost::urls::format(
+        "/redfish/v1/Chassis/{}/Assembly#/Assemblies/{}", chassisId,
+        std::to_string(it - assemblyList.begin()));
+
+    asyncResp->res.jsonValue["Slots"][index]["Links"]["Oem"]["IBM"]
+                            ["@odata.type"] = "#OemPCIeSlots.v1_0_0.PCIeLinks";
+    asyncResp->res.jsonValue["Slots"][index]["Links"]["Oem"]["IBM"]
+                            ["AssociatedAssembly"] = std::move(item);
+}
+
+inline void afterLinkAssociatedDiskBackplane(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, size_t index,
+    const boost::system::error_code& ec,
+    const dbus::utility::MapperEndPoints& endpoints)
+{
+    if (ec)
+    {
+        if (ec.value() == EBADR)
+        {
+            // Disk backplane association not found for this pcie slot.
+            BMCWEB_LOG_DEBUG("Disk backplane association not found");
+            return;
+        }
+        BMCWEB_LOG_ERROR("DBUS response error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    if (endpoints.empty())
+    {
+        BMCWEB_LOG_DEBUG("No association was found for disk backplane drive");
+        messages::internalError(asyncResp->res);
+        return;
+    }
+
+    // Each slot points to one disk backplane, so picking the top one
+    // or the only one we will have instead of looping through.
+    std::string drivePath = endpoints[0];
+    std::string chassisId = "chassis";
```

Comment:
endpoints() was created in dbus method and I was wondering whether it can still survive over async-calls, but it may be okay.  I'll change to `const std::string&`.
chassid should be no differences as it is allocated here anyway.